rye/src/cli/publish.rs

```diff
@@ -8,22 +8,27 @@ use age::{
 };
 use anyhow::{bail, Context, Error};
 use clap::Parser;
-use toml_edit::{Item, Table};
+use toml_edit::{Document, Item, Table};
 use url::Url;
 
 use crate::bootstrap::ensure_self_venv;
 use crate::platform::{get_credentials, write_credentials};
 use crate::pyproject::PyProject;
 use crate::utils::{escape_string, get_venv_python_bin, tui_theme, CommandOutput};
 
+const DEFAULT_USERNAME: &str = "__token__";
+const DEFAULT_REPOSITORY: &str = "pypi";
+const DEFAULT_REPOSITORY_DOMAIN: &str = "upload.pypi.org";
+const DEFAULT_REPOSITORY_URL: &str = "https://upload.pypi.org/legacy/";
+
 /// Publish packages to a package repository.
 #[derive(Parser, Debug)]
 pub struct Args {
     /// The distribution files to upload to the repository (defaults to <workspace-root>/dist/*).
     dist: Option<Vec<PathBuf>>,
     /// The repository to publish to.
-    #[arg(short, long, default_value = "pypi")]
-    repository: String,
+    #[arg(short, long)]
+    repository: Option<String>,
```

Comment:
Resolution of credentials and the repository configuration can get messy. So defaulting to `"pypi"` here only made it more difficult to to write something easier to follow.

This is now `Option<String>` so that it's possible to pass *'no repository'* onto the rest of the logic. Keyring fallback doesn't require a repository. It requires a repository url and a username.

Reply:
A question I'd have would be if we should default to 'pypi' for reasonable cases. Like if token is provided but no credentials file is found.